### DISCUSSION TITLE
Client Side API Key

### DISCUSSION THREAD

**Discussion by wacioc** [deepgram/community#598] [General help] [answered]
We'd like to decouple our back-end from the live-streaming process since that involves maintaining 2 websockets for each ongoing recording:
- 1 for the client -> server audio data
- 1 for the server -> Deepgram

For obvious reasons we don't want to share a universal API key on client side since anyone can grab it and exploit it.

I know we can create API keys that expire using https://developers.deepgram.com/docs/create-additional-api-keys , but I'm curious if this is scalable where we would generate a temporary API key for each client recording so the client can directly stream the data to Deepgram.
Assembly AI has this feature where you can create temporary tokens that can be used to authorise clients to use the API for transcribing their audio.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
Creating a temporary API key per session is exactly what I would do given the constraints you are mentioning, it's one of the use-cases that we anticipate clients doing.

**Reply by wacioc** [reply]
Does the API key need to be valid for the entire lifespan of the web socket or only when initialising it? Trying to figure out what would happen if the API key expires while the user is recording.

**Reply by jjmaldonis** [reply]
Hey @wacioc, the API only needs to be valid when the websocket is initially connected.

**Comment by Chang-Hai** [comment]
所以，到底怎么在服务端生成临时token给客户端，然客户端拿这个临时token建立一个实时语音识别。我好像没有在文档里找到方法
